### PR TITLE
feat: update readme, add jsdoc annotations, fix interfaces, formatting

### PR DESCRIPTION
**Pull request by Demian222**
BREAKING CHANGE: types refactoring

### CODE REVIEW

File: packages/connection-provider/src/main/ts/interfaces.ts [hunk 1]
```diff
@@ -17,15 +16,6 @@ export type IConsulDiscovery = {
   serviceName: string
 }
 
-export type IConnectionParams = {
```
Comment (antongolub) [inline packages/connection-provider/src/main/ts/interfaces.ts:20]: 1. Types refactoring will bring a breaking change
2. Consul module _implements_ conn provider contract, so the types should stay here or be moved upper to the common module.

**Review by antongolub — CHANGES_REQUESTED** [review]

File: packages/logger/src/main/ts/app.pipe.ts [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
-    version,
-    host,
-  })
-  return entry
-}
+/**
+ * @param {string} name - app name
+ * @param {string} version - app version
+ * @param {string} host - hostname
+ * @return {ILogEntry} assign logger entry meta and params (name, versions, host)
```
Comment (oljekechoro) [inline packages/logger/src/main/ts/app.pipe.ts:7]: assigned, imho?

File: packages/logger/src/main/ts/winston.ts [hunk 3]
```diff
@@ -12,13 +12,17 @@ export type TWinstonEntry = {
...
-  format: { combine, printf, timestamp } ,
+  transports: { Console, File },
+  format: { combine, printf, timestamp },
 }: any = winston
 
 const isTimestampValid = (timestamp: any) =>
   typeof timestamp === 'number' || Date.parse(timestamp)
 
+/**
+ * @param {TWinstonEntry} entry - logEntry with format winston
```
Comment (oljekechoro) [inline packages/logger/src/main/ts/winston.ts:23]: winston format

File: packages/logger/src/main/ts/winston.ts [hunk 4]
```diff
@@ -12,13 +12,17 @@ export type TWinstonEntry = {
...
+ * @return - logEntry with format kibana
```
Comment (oljekechoro) [inline packages/logger/src/main/ts/winston.ts:24]: kibana format

File: packages/consul/src/main/ts/consul.service.ts [hunk 5]
```diff
@@ -8,15 +8,41 @@ import { IConfig, ILogger, IPromise } from '@qiwi/substrate'
...
+   */
+  getConnectionParams(
+    serviceName: string,
+  ): Promise<IConnectionParams | undefined>
+
+  /**
+   * get value by key from storage
+   *
+   * @param key
+   * @return Return value by key
```
Comment (oljekechoro) [inline packages/consul/src/main/ts/consul.service.ts:36]: value by key

File: packages/consul/src/main/ts/consul.service.ts [hunk 6]
```diff
@@ -8,15 +8,41 @@ import { IConfig, ILogger, IPromise } from '@qiwi/substrate'
...
+  /**
+   * register service in consul
+   */
+  register(): IPromise
+
+  /**
+   * get connection params by service name
+   *
+   * @param serviceName
+   * @return Return host and port to connection
```
Comment (oljekechoro) [inline packages/consul/src/main/ts/consul.service.ts:26]: host and port for connection

**Review by oljekechoro — CHANGES_REQUESTED** [review]
small typos

File: package.json [hunk 7]
```diff
@@ -10,7 +10,7 @@
     "clean": "yarn workspaces foreach -p run clean",
     "prebuild": "yarn infra:tsc -b packages/facade/tsconfig.es6.json && yarn buildstamp",
     "buildstamp": "yarn infra:buildstamp --git --output='buildstamp.json'",
-    "build": "yarn clean && yarn prebuild && yarn workspaces foreach -pt run build",
+    "build": "yarn clean && yarn prebuild && yarn workspaces foreach --topological-dev run build",
```
Comment (oljekechoro) [inline package.json:13]: -p?

File: packages/connection-provider/src/main/ts/connection-provider.service.ts [hunk 8]
```diff
@@ -1,39 +1,39 @@
...
+    serviceProfile: IServiceDeclaration,
+  ): Promise<IConnectionParams | undefined>
 }
 
 @Injectable()
 export class ConnectionProviderService implements IConnectionProvider {
   constructor(
     @Inject('IConsul')
-    private consul: IConsulService,
+    private consul: IAbstractConsulService,
```
Comment (oljekechoro) [inline packages/connection-provider/src/main/ts/connection-provider.service.ts:26]: IDiscoverable?

File: packages/thrift/src/main/ts/thrift.client.ts [hunk 9]
```diff
@@ -34,38 +35,49 @@ export class ThriftClientProvider implements IThriftClientProvider, OnModuleDest
...
+    serviceProfile: IThriftServiceProfile | string,
+  ): Promise<IConnectionParams & { thriftServiceName: string } > {
+    const profile = this.getServiceProfile(serviceProfile)
+    const connectionParams = await this.connectionProvider.getConnectionParams(
+      profile,
+    )
+    if (!connectionParams) {
+      throw new Error(
+        `Bad connections params - ${
+          typeof serviceProfile === 'string' ? serviceProfile : ''
```
Comment (oljekechoro) [inline packages/thrift/src/main/ts/thrift.client.ts:67]: а тут типа при ошибке, если не string, будет `Bad connections params - `?

**Review by oljekechoro — APPROVED** [review]

**Pull request merged**